source/beamline/REIXS/REIXSSampleMotor.cpp

```diff
@@ -0,0 +1,377 @@
+#include "REIXSSampleMotor.h"
+
+#include "beamline/CLS/CLSMDriveMotorControl.h"
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "util/AMErrorMonitor.h"
+
+REIXSSampleMotor::REIXSSampleMotor(AMMotorGroupObject::MotionDirection direction,
+                                   const QString& name,
+                                   const QString& units,
+		                           AMControl* horizontalTranslationControl,
+		                           AMControl* normalTranslationControl,
+		                           AMControl* verticalRotationControl,
+		                           QObject *parent,
+		                           const QString& description) :
+
+    AMPseudoMotorControl(name, units, parent, description)
+{
+	direction_ = direction;
+
+	if(direction_ == AMMotorGroupObject::VerticalMotion) {
+
+		AMErrorMon::alert(this,
+		                  REIXS_SAMPLE_MOTOR_INVALID_DIRECTION,
+		                  QString("REIXS Sample Motor initialized in vertical axis"));
+	}
+
+	horizontalTranslationControl_ = horizontalTranslationControl;
+	normalTranslationControl_ = normalTranslationControl;
+	verticalRotationControl_  = verticalRotationControl;
+
+	angleOffset_ = 0;
+	connectedOnce_ = false;
+
+	addChildControl(horizontalTranslationControl_);
+	addChildControl(normalTranslationControl_);
+	addChildControl(verticalRotationControl_);
+
+	connect(verticalRotationControl_, SIGNAL(valueChanged(double)),
+	        this, SLOT(onVerticalAxisRotated(double)));
+}
+
+
+double REIXSSampleMotor::angleOffset() const
+{
+	return angleOffset_;
+}
+
+double REIXSSampleMotor::totalRotation() const
+{
+	if(connected_) {
+
+		double combinedRotation = verticalRotationControl_->value() + angleOffset_;
+
+		while(combinedRotation > 360) {
+			combinedRotation -= 360;
+		}
```

Comment:
this sounds like a `fmod()` function call, like  `combinedRotation = fmod(combinedRotation, 360);`. I'm not sure whether we can use `fmod()` in qt easily.